x-pack/plugins/security/public/usage_collection/index.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Subscription } from 'rxjs';
+
+import type { HttpStart } from 'src/core/public';
+
+import type { AuthenticatedUser, SecurityLicense } from '../../common';
+
+interface SecurityUsageCollectionSetupParams {
+  securityLicense: SecurityLicense;
+}
+
+interface SecurityUsageCollectionStartParams {
+  http: HttpStart;
+  getCurrentUser: () => Promise<AuthenticatedUser>;
+}
+
+interface SecurityUsageCollectionAuthType {
+  username_hash: string;
+  timestamp: number;
+  auth_type: string;
+}
+
+export class SecurityUsageCollectionService {
+  public static KeyAuthType = 'kibana-user-auth-type';
+  private securityLicense!: SecurityLicense;
+  private securityFeaturesSubscription?: Subscription;
+
+  public setup({ securityLicense }: SecurityUsageCollectionSetupParams) {
+    this.securityLicense = securityLicense;
+  }
+
+  public async start({ http, getCurrentUser }: SecurityUsageCollectionStartParams) {
+    // wait for the user to be authenticated before doing UsageCollection
+    this.securityFeaturesSubscription = this.securityLicense.features$.subscribe(
+      ({ allowRbac }) => {
+        if (allowRbac) {
+          getCurrentUser().then(() => this.recordAuthTypeUsageCollection(http));
```

Comment:
What do we do in the event of a promise rejection? I think it's fine to ignore it, but we probably want to explicitly ignore it with a `.catch` block.